storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTask.java

```diff
@@ -353,4 +358,10 @@ public void close() {
             }
         }
     }
+
+    public Set<TopicPartition> metadataPartitionsAssigned() {
+        return assignedMetaPartitions.stream()
```

Comment:
Can we return an immutable `Set` of `assignedMetaPartitions` instead of topic-partitions as the topic names are repetitive? This is currently being passed to build a string of partitions in an Exception message [here](https://github.com/apache/kafka/pull/14045/files#diff-7d10de4e1fdb683a452acb203d1ee92b64539a2b0b19533a6abc42fa97b2f04aR115).

Reply:
Sure, applying suggestion